### PR TITLE
Add Utilities Tests (for Path)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

This PR adds tests for https://github.com/NREL/OpenStudio/issues/3436, with PR https://github.com/NREL/OpenStudio/pull/3515

This Pull Request is concerning:

 - [ ] **Case 1 - `NewTest`:** a new test for a new model API class,
 - [ ] **Case 2 - `TestFix`:** a fix for an existing test. The GitHub issue should be referenced in the PR description
 - [ ] **Case 3 - `NewTestForExisting`:** a new test for an already-existing model API class
 - [x] **Case 4 - `Other`:** Something else, like maintenance of the repo, or just committing test results with a new OpenStudio version.

Depending on your answer, please fill out the required section below, and delete the three others.
Leave the review checklist in place.

**Note**: You can open a specific PR template directly by appending `?template=xxx` argument with the value `newtest.md`, `testfix.md` or `newtestforexisting.md`.

----------------------------------------------------------------------------------------------------------

### Case 4: Other

Please be as explicit as possible about the changes you have made, and why they are warranted.

----------------------------------------------------------------------------------------------------------

### Review Checklist

 - [ ] Code style (indentation, variable names, strip trailing spaces)
 - [ ] Functional code review (it has to work!)
 - [ ] Matching OSM test has been added or `# TODO` added to `model_tests.rb`
 - [ ] Appropriate `out.osw` have been committed
 - [ ] Test is stable
 - [ ] The appropriate labels have been added to this PR:
   - [ ] One of: `NewTest`, `TestFix`, `NewTestForExisting`, `Other`
   - [ ] If `NewTest`: add `PendingOSM` if needed

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,9 +16,11 @@ site_kbtu_pct_change*.png
 
 # Ignore the out.osw files that have a custom tag (something like '_whatever' between 'out' and '.osw')
 test/*out_*.osw
+test/*out_*.status
```
Comment (macumber) [inline .gitignore]: Probably ok to do these tests here but I didn't know if you were aware of the tests here: https://github.com/NREL/OpenStudio/tree/develop/openstudiocore/ruby/test and here https://github.com/NREL/OpenStudio/tree/develop/openstudiocore/src/cli/test ?

Especially this one: https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/cli/test/test_encodings.rb

Comment (macumber) [reply .gitignore]: I think the unit tests in OpenStudio could be improved by calling them with both the CLI *and* the system ruby with -I pointing to the built bindings (not done currently):

https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/cli/CMakeLists.txt#L330

Comment (macumber) [reply .gitignore]: Looks like we aren't calling the tests in /openstudiocore/ruby/test anyone, we should re-enable those.  They should be called with both the CLI and system ruby:

https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/cli/CMakeLists.txt#L330

Comment (macumber) [reply .gitignore]: I wonder if these issues are related to https://github.com/NREL/openstudio-extension-gem/issues/12

Comment (jmarrec) [reply .gitignore]: I wasn't aware of these tests in OpenStudio itself now. Do you want me to move stuff over there?
I guess the only thing that would make it more interesting to do it here is that we could keep track of a history of encodings (which I record, and aim to do so **by platform**) and success/failures.

Comment (macumber) [reply .gitignore]: I don't think we can test this stuff enough, so I would be in favor of duplicate tests in both places.

**Pull request merged**